src/boolean3.cpp

```diff
@@ -247,46 +167,26 @@ struct Kernel11 {
       const double start2 = la::dot(diff, diff);
       diff = vertPosP[p1e] - vec3(xyzz11);
       const double end2 = la::dot(diff, diff);
-      const double dir = start2 < end2 ? normalP[p1s].z : normalP[p1e].z;
+      const double dir = start2 < end2 ? normal[p1s].z : normal[p1e].z;
 
       if (!Shadows(xyzz11.z, xyzz11.w, expandP * dir)) s11 = 0;
     }
-  }
-};
 
-std::tuple<Vec<int>, Vec<vec4>> Shadow11(SparseIndices &p1q1,
-                                         const Manifold::Impl &inP,
-                                         const Manifold::Impl &inQ,
-                                         double expandP) {
-  ZoneScoped;
-  Vec<int> s11(p1q1.size());
-  Vec<vec4> xyzz11(p1q1.size());
-
-  for_each_n(autoPolicy(p1q1.size(), 1e4), countAt(0_uz), p1q1.size(),
-             Kernel11({xyzz11, s11, inP.vertPos_, inQ.vertPos_, inP.halfedge_,
-                       inQ.halfedge_, expandP, inP.vertNormal_, p1q1}));
-
-  p1q1.KeepFinite(xyzz11, s11);
-
-  return std::make_tuple(s11, xyzz11);
+    return std::make_pair(s11, xyzz11);
+  }
 };
 
 struct Kernel02 {
-  VecView<int> s;
-  VecView<double> z;
   VecView<const vec3> vertPosP;
```

Comment:
Maybe we should change the name of these P and Q, because we will swap them. Not sure what is the best name here.

Reply:
I like your renaming of them to e.g. `vertPosA` and `halfedgeB` to handle the swapping.